DirectProgramming/DPC++FPGA/include/utils.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+/*
+  Static implementation of the base 2 logarithm function
+*/
+template <typename T>
+static constexpr T Log2(T n) {
+  T ret = T(0);
+  T val = n;
+  while (val > T(1)) {
+    val >>= 1;
+    ret++;
+  }
+  return ret;
+}
+
+/*
+  Static implementation of the CEIL base 2 logarithm function
+*/
+template <unsigned int n, uint8_t remains = 0>
```

Comment:
I think @tyoungsc has an include file you can use that defines these constexpr math functions, rather than recreating them here?